### PR TITLE
Feat/242 add input phone

### PR DESCRIPTION
**Pull request by tleperou**
# What is it?

- [x] Feature / enhancement
- [ ] Bug
- [x] Docs / tests

# Description

Input Phone aims at formatting and validating with ease international phone number.

## In details

1. automatic detection of country code from phone number input
2. manual country code from select
3. validate: 'INVALID_COUNTRY' | 'NOT_A_NUMBER'  | 'TOO_SHORT' | 'TOO_LONG' | 'INVALID_LENGTH' | 'MAYBE_VALID' | 'NOT_VALID' | 'VALID'

## Accessibility

- Native HTML element
- Fully keyboard control
- Switch countries list and input number using CSS only
- Axe tested

**Note**: Shall integrate the headless Select, maybe when the autocompletion is implemented + Axe tested.

# Use cases and why

Any where a phone number is required

# Screenshots/Demo

image
image

# Checklist:

- [x] My code follows the [developer guidelines of this project](https://github.com/qwikifiers/qwik-ui/blob/main/CONTRIBUTING.md)
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [ ] Added new tests to cover the fix / functionality

### CODE REVIEW

**Comment by tleperou** [comment]
Accessibility test with Axe

image

image

**Comment by gioboa** [comment]
That's great. Can you describe how to test the component with axe please?
You can add the steps here CODING_STANDARDS.md

**Comment by tleperou** [comment]
I have read the CLA Document and I hereby sign the CLA

recheck

**Comment by tleperou** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by tleperou** [comment]
I proceed with the code standard updates in a separate MR, #244.

Shall tests results be associated within the component, somehow ?

File: apps/website/package.json [hunk 1]
```diff
@@ -1,5 +1,8 @@
 {
   "name": "@qwik-ui/website",
   "description": "The Qwik UI Website",
-  "private": true
+  "private": true,
+  "devDependencies": {
+    "libphonenumber-js": "1.10.24"
```
Comment (gioboa) [inline apps/website/package.json:6]: Can you move this dependency in the root package.json please?

File: packages/headless/package.json [hunk 2]
```diff
@@ -34,5 +34,9 @@
   },
   "dependencies": {
     "@floating-ui/dom": "1.0.10"
+  },
+  "devDependencies": {
+    "country-list-json": "1.1.0",
```
Comment (gioboa) [inline packages/headless/package.json:39]: Can you move these deps in the root package.json please?

**Comment by gioboa** [comment]
a11y need to be documented so I think we can create a markdown with our components + axe screenshots

**Comment by tleperou** [comment]
!245 for the coding standard.

I mentioned the way to store test reports: in Github or discord (temporarily).

I better dig further within the extension to see who to collocate the report nicely with the component

**Comment by tleperou** [comment]
> a11y need to be documented so I think we can create a markdown with our components + axe screenshots

Added in a `.json` format.

The structure result refers to the language. It seemed to me be not enough relevant to the component's scope to fix the issues of the structure.

**Pull request merged**